Sources/TensorFlow/Operators/Math.swift

```diff
@@ -2656,6 +2722,13 @@ public extension Tensor where Scalar: TensorFlowFloatingPoint {
     @inlinable
     @differentiable(wrt: self)
     func moments(squeezingAxes axes: Tensor<Int32>) -> Moments<Scalar> {
+        precondition(axes.rank == 1, "Axes must have rank 1")
```

Comment:
Please move `precondition(axes.rank == 1, ...)` into the body of `Tensor.areAxesInRange(_: Tensor<Int32>)`. Same elsewhere.